packages/@ember/-internals/runtime/lib/system/array_proxy.js

```diff
@@ -11,10 +11,13 @@ import {
   removeArrayObserver,
   replace,
 } from '@ember/-internals/metal';
+import { symbol } from '@ember/-internals/utils';
 import EmberObject from './object';
 import { isArray, MutableArray } from '../mixins/array';
 import { assert } from '@ember/debug';
 
+const CONSUMED = symbol('ARRAY_PROXY_CONSUMED');
```

Comment:
Is there a reason to use this over a `WeakMap`?

Reply:
Not really, we could definitely update it to a `WeakMap` or `WeakSet`. I figured there _may_ be perf implications, but don't have a strong opinion one way or the other

Reply:
Ya, I generally agree with @chadhietala here. I'd prefer to avoid adding arbitrary fields to the underlying objects. I doubt there is much difference between this implementation and a WeakMap/WeakSet based one, but I'm happy to test if you'd like to assuage concerns...